### ISSUE TITLE
Harden tests of Triangular Treasure

### ISSUE THREAD

**Issue by hobovsky** [codewars/content-issues#194]
As a follow-up of the deduplication issue #188, the kata [Triangular Treasure](https://www.codewars.com/kata/525e5a1cb735154b320002c8) should have its tests updated to allow only O(1) solutions.

List of languages to change:

- [ ] C
- [x] COBOL
- [x] CoffeeScript
- [ ] C# (no random tests)
- [ ] Dart
- [ ] Haskell
- [ ] Java
- [x] JavaScript
- [ ] Julia
- [ ] NASM
- [ ] Ruby
- [ ] Rust
- [ ] SQL (O(n) solutions are more difficult than O(1) solutions)
- [ ] Swift (no random tests)

**Comment by monadius** [comment]
I am not sure how to update C, C#, NASM, Rust. They all require 32-bit integer results. It is not reasonable to enforce O(n) solutions with this return type (more than 200_000 tests are required in NASM to eliminate O(n) solutions). Java already uses `long` for results. I think 64-bit integers should be used in all statically typed languages. Do you agree?

**Comment by hobovsky** [comment]
Yes, I think switching to 64 bit type is OK. For a brief moment i even considered testing for inputs which would trigger overflow if multiplied first and divided later, but I dont know if this would be still fair for 7 kyu.

**Comment by akar-0** [comment]
Unless I am doing something bad or missing something, it seems impossible to make iterative solutions fail in Rust, even using i64, probably due to some compiler optimization, see [this fork](https://www.codewars.com/kumite/6287684a2d911d02b2f559f1?sel=653bb71c386565655b4d7d1c) (random test with `n = i32::MAX as i64`, reference solution is iterative).

**Comment by monadius** [comment]
The Rust compiler knows how to compute the sum of a range efficiently. So the compiled code does not contain any loops and your solution is not iterative. Naive solutions (with `for` or `while` loops) are iterative and they will fail your tests if the exponent `p` is increased in random tests. See this Godbolt examples: https://godbolt.org/z/jYv38bs6Y

I forked your kumite and ran some tests. It seems that the range sum is not optimized on Codewars (which is strange because the [docs](https://docs.codewars.com/languages/rust/) show `opt-level = 1`). So it is enough to generate large values of `n` to fail iterative solutions:
```rust
let n = rng.gen_range(0..10_i64.pow(9));
```

**Comment by akar-0** [comment]
Ok, thanks, I didn't know that.

**Comment by KayleighWasTaken** [comment]
[Updated CoffeeScript tests.](https://www.codewars.com/kumite/6544ac9588a141001caab4ce?sel=6544ac9588a141001caab4ce)
